### PR TITLE
Feature/tao 6219 sync test takers

### PR DESCRIPTION
**Pull request by krampstudio**
This story is the first one in the _synchronization serie_ : the synchronization of test takers.

## How to test the story

On the endpoint 
 - open any webbrowser on the endpoint : http://spielplatz.taocloud.org/app-central-sync
 - login the backoffice (ask me an account)
 - create a Sync Manager user
 - check or add some test takers
 
On the app (or using the dev server `npm run dev` -> `http://localhost:3000`) : 
 - login with the created Sync Manager
 - Start the synchronization
 - All test takers should be included
 - Logout and try to login with a test taker

Go back to the endpoint  (could be done in mulitple times ) : 
 - remove some test takers
 - update some test takers 
 - add new test takers 

On the app ; 
 - login with the  Sync Manager
 - Start the synchronization
 - Check if you can still login with the removed account, the new accounts, etc.

## PR details
 - Some refactoring especially on the `request` side to inherit the API config
 - Add some new `app/core` components
 - Even if the task didn't require it explicitly, I've created a "temporary" layout and the controllers in order to test it (the UX will be adapted in a next story)
 - I've also added a "wipeout" feature to clear the app data

### CODE REVIEW

File: src/js/app/service/synchronization/client.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+ * should be used for a given synchronization type (see {@link app/service/synchronization/testTaker}).
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'module',
+    'urlParser',
```
Comment (jsconan) [inline src/js/app/service/synchronization/client.js]: seems to be unused

File: src/js/app/service/synchronization/testTaker.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    'app/service/synchronization/client'
+], function(_, loggerFactory, promiseQueueFactory, userService, userDataMapper, syncClientFactory){
+    'use strict';
+
+    var logger  = loggerFactory('app/service/synchronization/testTaker');
+
+    var resourceType = 'test-taker';
+
+    /**
+     * we will load resouces by batch of 10
```
Comment (jsconan) [inline src/js/app/service/synchronization/testTaker.js]: small typo

File: src/js/app/core/digest.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     * @returns {String} the hex representation of the buffer
+     */
+    var bufferToHexString = function bufferToHexString(buffer) {
+        return [].map.call(new Uint8Array(buffer), function(val){
+            return  ('00' + val.toString(16)).slice(-2);
+        }).join('');
+    };
+
+    /**
+     * Create a hash/checskum from a given string
```
Comment (jsconan) [inline src/js/app/core/digest.js]: small typo

File: src/js/test/component/wipeout/test.js [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+            })
+            .on('wipeout', function(){
+                assert.ok(true, 'Wipeout called');
+                assert.ok(this.is('waiting'), 'The component remains in waiting state!');
+
+                QUnit.start();
+            });
+    });
+
+    QUnit.asyncTest('Cancel wipeout', function(assert) {
```
Comment (jsconan) [inline src/js/test/component/wipeout/test.js]: This test is failing

File: src/js/app/component/synchronizer/scss/synchronizer.scss [hunk 5]
```diff
@@ -0,0 +1,106 @@
+@import "inc/bootstrap";
+
+@keyframes rotate {
+  0% {
+      transform : rotate(0deg);
+  }
+  90% {
+      transform : rotate(360deg);
```
[the remark this answers is not in the document]
Comment (krampstudio) [reply src/js/app/component/synchronizer/scss/synchronizer.scss]: Based on the targeted platforms, this property is supported by the webview. I was wondering if it was necessary to generate the whole fallback chain

File: src/js/app/service/authentication/syncManagerAdapter.js [hunk 6]
```diff
@@ -27,81 +27,62 @@ define([
...
+            if(!credentials || _.isEmpty(credentials.username) || _.isEmpty(credentials.password)){
+                return Promise.resolve({
+                    success: false,
+                    message: errorMessages.unauthorized
+                });
+            }
+            return request({
+                data : JSON.stringify({
+                    login:    credentials.username,
+                    password: credentials.password
```
[the remark this answers is not in the document]
Comment (krampstudio) [reply src/js/app/service/authentication/syncManagerAdapter.js]: Yes, that's the way taoSync works.

**Pull request merged**